estuary_updater/handlers/koji.py

```diff
@@ -0,0 +1,90 @@
+# SPDX-License-Identifier: GPL-3.0+
+
+from __future__ import unicode_literals
+
+from estuary.models.koji import KojiBuild, KojiTag
+
+from estuary_updater.handlers.base import BaseHandler
+from estuary_updater import log
+
+
+class KojiHandler(BaseHandler):
+    """A handler for Koji related messages."""
+
+    @staticmethod
+    def can_handle(msg):
+        """
+        Determine if this handler can handle this message.
+
+        :param dict msg: a message to be analyzed
+        :return: a bool based on if the handler can handle this kind of message
+        :rtype: bool
+        """
+        supported_topics = [
+            '/topic/VirtualTopic.eng.brew.build.complete',
+            '/topic/VirtualTopic.eng.brew.build.tag',
+            '/topic/VirtualTopic.eng.brew.build.untag'
+        ]
+        return msg['topic'] in supported_topics
+
+    def handle(self, msg):
+        """
+        Handle a message and update Neo4j if necessary.
+
+        :param dict msg: a message to be processed
+        """
+        topic = msg['topic']
+
+        if topic == '/topic/VirtualTopic.eng.brew.build.complete':
+            self.build_complete_handler(msg)
+        elif topic == '/topic/VirtualTopic.eng.brew.build.tag' or \
+                      '/topic/VirtualTopic.eng.brew.build.untag':
+            self.build_tag_handler(msg)
+        else:
+            raise RuntimeError('This message is unable to be handled: {0}'.format(msg))
+
+    def build_complete_handler(self, msg):
+        """
+        Handle a build complete message and update Neo4j if necessary.
+
+        :param dict msg: a message to be processed
+        """
+        build_info = {
+            'id': msg['body']['msg']['info']['id'],
+            'epoch': msg['body']['msg']['info']['epoch'],
+            'state': msg['body']['msg']['info']['state'],
+            'creation_ts': msg['body']['msg']['info']['creation_ts'],
+            'start_ts': msg['body']['msg']['info']['start_ts'],
+            'completion_ts': msg['body']['msg']['info']['completion_ts'],
+            'extra': msg['body']['msg']['info']['extra'],
+            'package_name': msg['body']['msg']['info']['name'],
+            'version': msg['body']['msg']['info']['version'],
+            'release': msg['body']['msg']['info']['release'],
+            'owner_name': msg['body']['msg']['info']['owner_name']
+        }
+
+        self.get_or_create_build(msg['body']['msg']['info']['id'], build_info=build_info,
+                                 original_nvr=msg['body']['msg']['info']['nvr'])
```

Comment:
`original_nvr` is only a value for container builds built by Freshmaker. Drop that from the method call and it should be good afterwards.